src/cargo/ops/cargo_compile.rs

```diff
@@ -404,12 +404,6 @@ pub fn create_bcx<'a, 'cfg>(
                 .shell()
                 .warn("-Zbuild-std does not currently fully support --build-plan")?;
         }
-        if build_config.requested_kinds[0].is_host() {
-            // TODO: This should eventually be fixed. Unfortunately it is not
-            // easy to get the host triple in BuildConfig. Consider changing
-            // requested_target to an enum, or some other approach.
-            anyhow::bail!("-Zbuild-std requires --target");
-        }
```

Comment:
Please don't delete this, it's here for a reason. The `-Zbuild-std` unit graph is not correct unless `--target` is passed and a lot more is needed to get this working without it. Simply because the test suite doesn't fail doesn't mean this can be deleted.